core/src/impl/Kokkos_Utilities.hpp

```diff
@@ -202,6 +203,18 @@ template <typename T>
 constexpr bool dependent_false_v = !sizeof(T*);
 //==============================================================================
 
+template <typename T>
+std::enable_if_t<std::is_integral_v<T> && std::is_unsigned_v<T>, T>
+check_mul_of(T a, T b) {
```

Comment:
I don't like the function template name `check_mul_of` and would like to suggest we consider instead following existing practice like GCC builtins and introduce `add_overflow(x, y , &result) -> bool` or rename to `checked_mul` (maybe easier).

If you go for `checked_mul`, please do static assert rather than sfinae.

Reply:
Ok, `add_overflow` probably makes more sense looking at upcoming `<stdckdint.h>` / N2683.

Reply:
Live review suggestions:
- separate header in impl: something along the lines of `checked_integer_operations.h`
- SFINAE on integer
- static_assert( not unsigned, "not implemented yet")
- add unit test (!)